frontend/src/views/supp-allowances/SupplementaryFormView.vue

```diff
@@ -318,16 +408,34 @@ export default {
         applicationModel,
       )
     },
-    getModel(type) {
-      return this.models?.find((el) => el.supplementaryType === type)
+    getModel(type, currentTerm) {
+      return this.models?.find((el) => el.supplementaryType === type && el.renewalTerm == currentTerm)
     },
-    getTransportModels() {
-      return this.models?.filter((el) => el.supplementaryType === SUPPLEMENTARY_TYPES.TRANSPORT)
+    getTransportModels(term) {
+      return this.models?.filter((el) => el.supplementaryType === SUPPLEMENTARY_TYPES.TRANSPORT && el.renewalTerm == term)
     },
-    findAndUpdateModel(suppApplications, modelToUpdate) {
-      const found = suppApplications.find((application) => application.supplementaryType === modelToUpdate.supplementaryType)
+    findAndUpdateModel(suppApplications, modelToUpdate, renewalTerm) {
+      const found = suppApplications.find((application) => application.supplementaryType === modelToUpdate.supplementaryType && application.renewalTerm == renewalTerm)
```

Comment:
The name found always reads like a Boolean to me whereas it's actually the foundApplication/foundModel.

Reply:
changed!